third_party/move/move-compiler-v2/src/env_pipeline/lambda_lifter.rs

```diff
@@ -229,6 +234,18 @@ impl<'a> LambdaLifter<'a> {
                         name.display(env.symbol_pool())
                     ),
                 );
+                saw_error = true;
+            }
+            let param_abilities = ability_inferer.infer_abilities(&ty).1;
+            if !param_abilities.has_copy() {
+                env.error(
+                    &loc,
+                    &format!(
+                        "captured variable `{}` must have a value with `copy` ability", // TODO(LAMBDA)
```

Comment:
Is there a test case for this error?

Reply:
added `closure_args.move`, see `closure_args.lambda.exp`.

This exercises a lot of cases.